scintellometry/trials/crab/gpfinder.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+from __future__ import division, print_function
+
+import sys
+import numpy as np
+import astropy.units as u
+from astropy.time import Time
+import matplotlib.pyplot as plt
+
+MAX_RMS = 2.
+_fref = 610. * u.MHz  # ref. freq. for dispersion measure
+tsamp = 2.56e-4 # timestep in seconds
+thresh = 6. # detection threshold
+
+def rfi_filter_raw(raw):
+    #Simple RFI filter which zeros channels with high std dev
+    #Sum polarizations to get total intensity
+    if raw.shape[-1] == 4:
+        rawbins = raw[:, :, (0,3)].sum(-1, keepdims=True)
+    else:
+        rawbins = raw
+
+    std = rawbins.std(axis=0).mean()
+    raw_norm = rawbins/std
+    std_norm = raw_norm.std(axis=0,keepdims=True)
+    ok = std_norm < MAX_RMS
+    raw *= ok
+    return raw, ok
+
+
+def rfi_filter_power(power, t0):
+
+    # Detect and store giant pulses in each block through simple S/N test
+    if power.shape[-1] == 4:
+        freq_av = power[:, :, (0,3)].sum(1).sum(-1)
+    else:
+        freq_av = power.sum(1).squeeze()
+
+    sn = (freq_av-freq_av.mean()) / freq_av.std()
+    # Loop to remove outliers from S/N calculation
+    for i in xrange(3):        
+        back = np.argwhere(sn < thresh)
+        sn = (freq_av-freq_av[back].mean()) / freq_av[back].std()
+
+    peaks = np.argwhere(sn > thresh)
+
+    # Remove duplicates caused by pulse duration
+    pdiff = peaks - np.roll(peaks,1)
+    peaks = peaks[pdiff != 1]
+
+    f = open('giant_pulses.txt', 'a')
+
+    for peak in peaks:
+        tsr = Time((t0.unix + tsamp * peak), format='unix', scale='utc')
```

Comment:
This is can lead to round-off errors: just do `t0 + tsamp * peak` and define `tsamp` above including units.